### PR TITLE
Add Prettier to subroutine

### LINKED ISSUE #17: Tabs and spaces
We've been inconsistent about tabs vs. spaces, and we should decide which one to use.

### PR DESCRIPTION
**Pull request by iosenberg**
pls

### CODE REVIEW

**Review by AjaiKN — DISMISSED** [review]

**Comment by iosenberg** [comment]
I fucked up the commits, so this is no longer about updating my info and is now about adding Prettier as a subroutine lol

File: .htmlhintrc [hunk 1]
```diff
@@ -1,19 +1,18 @@
...
-    "id-unique": true,
-    "src-not-empty": true,
-    "attr-no-duplication": true,
-    "title-require": true,
-    "attr-unsafe-chars": true,
-    "html-lang-require": true,
-    "input-requires-label": true,
-    "tagname-specialchars": true,
-    "attr-no-unnecessary-whitespace": true,
-    "empty-tag-not-self-closed": true
```
Comment (wknowleskellett) [inline .htmlhintrc:18]: >    "empty-tag-not-self-closed": true

lol bye

File: README.md [hunk 2]
```diff
@@ -8,10 +10,8 @@ Follow this guide for help setting it up:
 https://docs.github.com/en/github-ae@latest/get-started/quickstart/set-up-git
 This is the hardest part, and we recommend doing so in club so that we can answer questions while you do this.
 
-After you have set up git, check to see whether you have Python installed on your computer by running in your terminal the following command:
-	> python --version
-If you receive an error, then you need to install python. Do so by running the following command in your terminal:
-	> python
+After you have set up git, check to see whether you have Python installed on your computer by running in your terminal the following command: > python --version
+If you receive an error, then you need to install python. Do so by running the following command in your terminal: > python
```
Comment (wknowleskellett) [inline README.md:17]: Quotes at the end of the line don't format properly. I understand prettier did this; would it maybe not do it if there were blank lines around the quotes? like this:

```
After you have set up git, do the thing:

	> python --version

If you receive an error, do the thing:

	> python
```

File: README.md [hunk 3]
```diff
@@ -21,15 +21,15 @@ At the top of this page is a "fork" button. Click this button to make a personal
...
 3. Open a terminal on your computer (e.g. Command Prompt for Windows)
 4. Navigate to where you want to store the repository locally
 5. Enter the following command in your terminal:
-	> git clone {the-link-you-copied}
+   > git clone {the-link-you-copied}
 6. Make a new branch (optional):
-	- enter the following commands:
-	> git checkout -b new-member
+   - enter the following commands:
+     > git checkout -b new-member
```
Comment (wknowleskellett) [inline README.md]: Why does this line agree with prettier?

**Comment by iosenberg** [comment]
@wknowleskellett Reformatted the code parts so the weird spacing should be fixed now

**Comment by AjaiKN** [comment]
People usually add the `node_modules` directory to `.gitignore`. The `package.json` and `package-lock.json` should be enough to recreate the `node_modules` directory using the `npm install` command.

**Comment by AjaiKN** [comment]
Actually, it looks like the `package-lock.json` doesn't match the `package.json`. I'm not sure how that would have happened. It also looks like the package-lock.json was created using an old version of npm.

**Comment by iosenberg** [comment]
@Aurelius333 
I can add `node_modules` to the `.gitignore` pretty easily. Would that require everyone who forks the repository to also run the `npm install` command then?
Also `package-lock.json` and `package.json` probably don't match because I wrote `package.json` and I don't know what I'm doing lol. Am I required to have two separate docs? or can I just add the info in `package.json` to the "husky" section of `package-lock.json?`
Not sure what to do about the out of date version either. This is the what was included in the most recent version of Prettier.

**Comment by AjaiKN** [comment]
>  I can add node_modules to the .gitignore pretty easily. Would that require everyone who forks the repository to also run the npm install command then?

Yes, they would have to do that to run prettier, but I think they would have to do that anyway to get the git pre-commit hooks to install.

> Not sure what to do about the out of date version either. This is the what was included in the most recent version of Prettier.

npm is a package manager, so you use npm to install Prettier, not the other way around.

> Also package-lock.json and package.json probably don't match because I wrote package.json and I don't know what I'm doing lol. Am I required to have two separate docs? or can I just add the info in package.json to the "husky" section of package-lock.json?

The idea is that you list what packages *you want* in your package.json. For example, it might be that you want version 2 of prettier. It doesn't matter to you whether it's version 2.1.0, 2.7.1, or whatever, you just want anything that's part of prettier version 2.

Then, you run `npm install`. When you do that, npm puts a ridiculous amount of stuff into node_modules. You don't generally commit node_modules because it's huge and creates ridiculous git diffs.

And when you run `npm install`, npm also automatically creates package-lock.json. That file shows the *exact versions* of *every* package, including all the dependencies of your dependencies. So even though you only care about prettier being version 2, package-lock.json records the fact that you have version 2.7.1. You *should* commit package-lock.json because then, when I run `npm install`, it'll see the package-json.lock, and I'll get the exact same versions that you have (in order to avoid inconsistencies between us). You shouldn't ever modify the package-lock.json yourself; npm will automatically update it for you.

So you should probably start from scratch by doing something like this:

1. Update npm if it's not already updated (https://docs.npmjs.com/try-the-latest-stable-version-of-npm)
2. Delete package-lock.json, since it somehow got out of date.
3. Run `npm install prettier`. That'll add prettier to your package.json and also to your package-lock.json.
4. If there are other packages you need (like husky), run `npm install husky` or whatever.

I hope that made a little sense. npm can be pretty confusing at first, so it might be easier to just wait until the next meeting.

Also, I think we'll probably want to add Prettier to our GitHub Actions.

**Comment by iosenberg** [comment]
Remade pull request in a way that works. Closing the cringe one.

**Pull request closed without merging**